CAMEL-13256: JOOQ component implementation with simple CRUD operation…

DenisIstomin opened the pull request:
…s support

oscerd commented on components/camel-jooq/pom.xml:
```diff
@@ -0,0 +1,101 @@
+    <description>Camel JOOQ support</description>
+
+    <properties>
+        <org.jooq.version>3.11.9</org.jooq.version>
```
This property placeholder can be in parent/pom

DenisIstomin replied:
Thanks, fixed

oscerd commented on components/camel-jooq/src/main/java/org/apache/camel/component/jooq/JooqConsumer.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Override
+    protected int poll() {
+        return 0;
```
Why this consumer always return 0? If the consumer doesn't do anything, it's better to avoid its support.

oscerd replied:
you can raise an UnsupportedOperationException in your createConsumer.

`throw new UnsupportedOperationException("You cannot receive messages from this endpoint");`

DenisIstomin replied:
Yes, consumer doesent do anything. I'll delete it. I'll implement it later with task CAMEL-13261.

dmvolod commented:
@DenisIstomin , looks good, but what do you think about to exclude automatic generated files from test and do this step with maven automatically. It will reduce JOOQ version upgrade time and notice about this step every update.

DenisIstomin commented:
@dmvolod thanks, that exactly what I did, when I started to implement this feature.
But then I changed my mind, because of:
1. Slower tests run
2. Broken intellisence in IDE, before running `maven compile` in console
I'll try to do something with that. Maybe I could generate thouse classes explicitly to `test/java/...` directory (not to 'target' directory).

dmvolod commented:
@DenisIstomin , good point, thanks.
How this tests were slowing? One, two or ten times? May be try to detect what the slowest point, maven, JOOQ platform itself or anything else?
It's not a good idea to generate files inside git tracked folders, it can be, but not always. For example we did it in camel-thrift, because thrift requires binary generator in testing env. For /target folder you could add it to the classpath in the IDE project, but it requires manually steps and of cause run maven compile before.

DenisIstomin commented:
1. Removed all generated JOOQ entity classes from git tracked folder
2. Added generation to `target/generated-test-sources/jooq`directory. Generation is starting only at `generate-test-sources` maven phase. It takes ~1 second.
3. Added 2 extra plugins to `parent/pom.xml`: `properties-maven-plugin`, `sql-maven-plugin`
@dmvolod @oscerd

oscerd commented:
I think it's safe to use this plugin, but if we are using them only in this component, it maybe makes sense to use that only in the component POM

oscerd commented:
@davsclaus @dmvolod WDYT?

dmvolod commented:
@oscerd , I agree with you, but looks like @DenisIstomin is talking about adding properties-maven-plugin and sql-maven-plugin versions to the parent/pom.xml. However, properties-maven-plugin already defined in it, but properties-maven-plugin-version not explicitly defined and passes as constant.

oscerd commented:
Then I think it's fine like this, sorry for this. I have 2000 tabs opened, I probably mixed stuff :-(

dmvolod commented:
No @oscerd , you are right, it's necessary to adopt properties-maven-plugin-version in other  components  using this plugin as @DenisIstomin introduced a new entity inside parent/pom.xml and we could missing or splitting versions in the future.

oscerd commented:
So maybe we can merge this one and then check for the plugins usage in the project

oscerd commented:
I'll merge this today

dmvolod commented:
@DenisIstomin , please rebase on actual master to reduce committers work on merge.

DenisIstomin commented:
Rebased, @dmvolod @oscerd

oscerd commented:
Thanks @DenisIstomin I'll merge tomorow

oscerd commented:
Ok, I merged the PR and fixed some minor Code style issues and a missing entry in a POM. Thanks for this @DenisIstomin :-) Really good job!
